### PR TITLE
fix: rate limit in dashboard and consent

### PR DESCRIPTION
**Pull request by siddhart1o1**

### CODE REVIEW

File: apps/dashboard/services/graphql/index.ts [hunk 1]
```diff
@@ -1,17 +1,36 @@
...
+  ApolloLink,
+} from "@apollo/client";
+import { registerApolloClient } from "@apollo/experimental-nextjs-app-support/rsc";
+
+const headerSettingLink = new ApolloLink((operation, forward) => {
+  const request = operation.getContext().request;
+  operation.setContext({
+    headers: {
+      "x-real-ip": request?.headers.get("x-real-ip") || "",
+      "x-forwarded-for": request?.headers.get("x-forwarded-for") || "",
```
Comment (bodymindarts) [inline apps/dashboard/services/graphql/index.ts:15]: do we really want to default to `""` - isn't it better to leave it undefined? cc @nicolasburtey

Comment (nicolasburtey) [reply apps/dashboard/services/graphql/index.ts:15]: agree but this may have to do with type?

File: apps/consent/app/graphql/apollo-config.ts [hunk 2]
```diff
@@ -1,15 +1,34 @@
...
+    headers: {
+      ...(authToken ? { authorization: `Bearer ${authToken}` } : undefined),
+    },
   });
 
+  const link = ApolloLink.from([headerSettingLink, httpLink]);
   return new ApolloClient({
     cache: new InMemoryCache(),
-    link: httpLink,
+    link: link,
```
Comment (nicolasburtey) [inline apps/consent/app/graphql/apollo-config.ts:32]:
```suggestion
    link,
```

File: apps/consent/services/galoy-auth/auth-instance.ts [hunk 3]
```diff
@@ -2,8 +2,25 @@ import { env } from "../../env";
...
+axiosInstance.interceptors.request.use(
+  (config) => {
+    const xRealIp = config.headers["x-real-ip"];
+    const xForwardedFor = config.headers["x-forwarded-for"];
+    if (xRealIp) {
+      config.headers["x-real-ip"] = xRealIp;
+    }
+    if (xForwardedFor) {
+      config.headers["x-forwarded-for"] = xForwardedFor;
+    }
```
Comment (nicolasburtey) [inline apps/consent/services/galoy-auth/auth-instance.ts:18]: this feels like a no-op? we' taking a value from `config.headers["x-real-ip"]` and redefining to the same value? same for `config.headers["x-forwarded-for"]`?

File: apps/dashboard/services/graphql/index.ts [hunk 4]
```diff
@@ -1,17 +1,36 @@
...
     return new ApolloClient({
       cache: new InMemoryCache(),
-      link: new HttpLink({
-        headers: {
-          ["Oauth2-Token"]: token,
-        },
-        uri: env.CORE_URL,
-        fetchOptions: { cache: "no-store" },
-      }),
+      link: link,
```
Comment (nicolasburtey) [inline apps/dashboard/services/graphql/index.ts:34]:
```suggestion
      link,
```

File: apps/consent/services/galoy-auth/auth-instance.ts (hunk 3 above)
Comment (siddhart1o1) [reply apps/consent/services/galoy-auth/auth-instance.ts:18]: yup this will not work need to add config

Comment (nicolasburtey) [reply apps/consent/services/galoy-auth/auth-instance.ts:18]: if you do `axiosInstance.interceptors.request.use(config => config)` this doesn't work?

File: apps/dashboard/services/graphql/index.ts (hunk 1 above)
Comment (siddhart1o1) [reply apps/dashboard/services/graphql/index.ts:15]: yes

File: apps/consent/services/galoy-auth/auth-instance.ts (hunk 3 above)
Comment (siddhart1o1) [reply apps/consent/services/galoy-auth/auth-instance.ts]: no

**Review by bodymindarts — APPROVED** [review]

**Pull request merged**